### PR TITLE
Reduxize auth0

### PR DESCRIPTION
**Pull request by meltedspork**

### CODE REVIEW

File: src/redux/modules/auth/__tests__/reducer.test.js [hunk 1]
```diff
@@ -64,10 +90,10 @@ describe('(Reducer) auth', () => {
   });
 
   it('should handle LOGOUT_SUCCESS', () => {
-    expect(authReducer(undefined, { type: authTypes.LOGOUT_SUCCESS, error: 'fakeError' })).to.deep.equal({
+    expect(authReducer(undefined, { type: authTypes.LOGOUT_SUCCESS, isLoggingOut: 1234, error: 'fakeError' })).to.deep.equal({
       isAuthenticated: false,
       isFetching: false,
-      isLoggingOut: false,
+      isLoggingOut: 1234,
```
Comment (kratsg) [inline src/redux/modules/auth/__tests__/reducer.test.js]: why 1234?

File: src/redux/modules/auth/actions.js [hunk 2]
```diff
@@ -29,28 +41,26 @@ export function logoutRequest() {
   };
 }
 
-export function logoutSuccess() {
+export function logoutSuccess({ logged }) {
   return {
     type: types.LOGOUT_SUCCESS,
+    action: {
+      isLoggingOut: !logged,
```
Comment (kratsg) [inline src/redux/modules/auth/actions.js]: I don't understand this. What is `logged`?

File: src/utils/AuthService.js [hunk 3]
```diff
@@ -27,33 +27,47 @@ export default class AuthService {
...
+      window.localStorage.removeItem(EXPIRES_AT);
+      window.localStorage.removeItem(PROFILE);
+      resolve(true);
+    });
   }
 
   static isAuthenticated() {
-    const token = AuthService.getIdToken();
-    return !!token && !AuthService.isTokenExpired(token);
+    const token = this.getIdToken();
```
Comment (kratsg) [inline src/utils/AuthService.js]: `this` isn't binded.

File: src/utils/AuthService.js [hunk 4]
```diff
@@ -27,33 +27,47 @@ export default class AuthService {
...
+    return !!token && !this.isTokenExpired(token);
```
Comment (kratsg) [inline src/utils/AuthService.js]: `this` isn't binded.

File: src/utils/AuthService.js [hunk 5]
```diff
@@ -85,7 +99,7 @@ export default class AuthService {
   }
 
   static getTokenExpirationDate() {
-    const token = AuthService.getIdToken();
+    const token = this.getIdToken();
```
Comment (kratsg) [inline src/utils/AuthService.js]: `this` isn't binded.

File: src/utils/AuthService.js [hunk 6]
```diff
@@ -97,12 +111,12 @@ export default class AuthService {
   }
 
   static isTokenExpired() {
-    const token = AuthService.getIdToken();
+    const token = this.getIdToken();
```
Comment (kratsg) [inline src/utils/AuthService.js]: `this` isn't binded.

File: src/utils/AuthService.js [hunk 7]
```diff
@@ -97,12 +111,12 @@ export default class AuthService {
...
     if (!token) {
       return true;
     }
 
-    const date = AuthService.getTokenExpirationDate();
+    const date = this.getTokenExpirationDate();
```
Comment (kratsg) [inline src/utils/AuthService.js]: `this` isn't binded.

File: src/redux/modules/auth/middleware.js [hunk 8]
```diff
@@ -0,0 +1,22 @@
+import AuthService from '../../../utils/AuthService';
+
+import { loginSuccess } from './actions';
+
+const authMiddleware = store => next => (action) => {
+  AuthService.handleAuthentication()
+    .then((logged) => {
```
Comment (kratsg) [inline src/redux/modules/auth/middleware.js]: What is `logged`? Should call this `authResult`.

File: src/redux/modules/auth/middleware.js [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+      window.location.href = '/';
+      store.dispatch(loginSuccess(logged));
```
Comment (kratsg) [inline src/redux/modules/auth/middleware.js]: What is `logged`? Should call this `authResult`.

File: src/redux/modules/auth/middleware.js [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+      next(action);
+    }).catch(() => {
+      // this is actually not true error
+      // console.log('err', err);
+      const isAuthenticated = AuthService.isAuthenticated();
+      if (isAuthenticated) {
+        console.log('getProfile: ', AuthService.getProfile());
```
Comment (kratsg) [inline src/redux/modules/auth/middleware.js]: `console.log` to be removed?

File: src/redux/modules/auth/__tests__/reducer.test.js (hunk 1 above)
Comment (meltedspork) [reply src/redux/modules/auth/__tests__/reducer.test.js]: Lol, i was trying to be funny. but yes that def should be boolean. Ill fix that.

File: src/utils/AuthService.js (hunk 3 above)
Comment (meltedspork) [reply src/utils/AuthService.js]: d'oh! you are right, i forgot to do `this.constructor.methodName`

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/static

File: src/redux/modules/auth/middleware.js (hunk 8 above)
Comment (meltedspork) [reply src/redux/modules/auth/middleware.js]: that was mistake on my part. You are right. should be labeled as `authResult`.

File: src/redux/modules/auth/middleware.js (hunk 10 above)
Comment (meltedspork) [reply src/redux/modules/auth/middleware.js]: we will remove this once we stabilize Auth0, i have updated to note this although

File: src/redux/modules/auth/actions.js (hunk 2 above)
Comment (meltedspork) [reply src/redux/modules/auth/actions.js]: See - https://github.com/deafchi/signsfive-web/blob/7c9645b1317baf1ae21e760b94a72973f8f3ffd9/src/redux/modules/auth/actions.js#L64 then to https://github.com/deafchi/signsfive-web/blob/7c9645b1317baf1ae21e760b94a72973f8f3ffd9/src/utils/AuthService.js#L28-L37

it has been converted into promise and returns true on resolve. But I realized this is badly naming. I will rename `logged` to `isLoggedOut`

**Comment by meltedspork** [comment]
Yeah, i considered to combine Submit-A-Sign and Sign-In logics into one component but problem with boottraps too many damn deep tags

**Comment by kratsg** [comment]
LOGIN_STATUS needs to be improved as well as handleAuthentication(). It seems like the state isn't changing correctly.

**Pull request merged**